src/WebJobs.Script/Description/DotNet/CSharp/Analyzers/InvalidFileMetadataReferenceAnalyzer.cs

```diff
@@ -0,0 +1,59 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Immutable;
+using System.IO;
+using System.Linq;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+
+namespace Microsoft.Azure.WebJobs.Script.Description.DotNet.CSharp.Analyzers
+{
+    [CLSCompliant(false)]
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    public sealed class InvalidFileMetadataReferenceAnalyzer : DiagnosticAnalyzer
+    {
+        private const string Title = "Invalid file metadata reference";
+        private const string MessageFormat = "The reference '{0}' is invalid, if you are attempting to add a framework reference, please remove the '{1}' file extension.";
```

Comment:
nit: add a period after "The reference '{0}' is invalid" (change comma to period).